link-grammar/linkage/linkage.c

```diff
@@ -875,6 +875,38 @@ const char * linkage_get_word(const Linkage linkage, WordIdx w)
 	return linkage->word[w];
 }
 
+// Need to free the returned value of this function later:
+// the memory pointed to by bare_word
+char * linkage_get_bare_word(const Linkage linkage, WordIdx w)
+{
+	if (!linkage) return NULL;
+	if (linkage->num_words <= w) return NULL; /* bounds-check */
+
+	// first look for unknown word tag
+	char delimiter = '[';
+	char *separator_ptr = strrchr(linkage->word[w], delimiter);
+
+	if(separator_ptr == NULL)
+	{
+	    // look for beginning of word tag
+		delimiter = '.';
+	    separator_ptr = strrchr(linkage->word[w], delimiter);
+	    if(separator_ptr == NULL)
+		{
+			// if no tags are aded, point to the end of the word
+			delimiter = '\0';
+	  		separator_ptr = strrchr(linkage->word[w], delimiter); // if there is no separator in the word
+		}
+	}
+	// copy and return word without the suffix
+	int position = separator_ptr - linkage->word[w];
+	char* bare_word = (char*) malloc((position + 1) * sizeof(char));
+	memcpy(bare_word, linkage->word[w], position);
```

Comment:
Hi @linas, I didn't notice Amir's update. Last time I read was his comment in google groups that his changes were still for an older version of link-grammar, so I decided to give it a try myself. My bad for not checking further commits.
Thanks for pointing out my mistakes, clearly my function is not addressing all possibilities. I will cancel my pull request.
About the memory leak part, I see that Amir uses alloca, which I didn't know before. So, in general will you have a memory leak if you malloc memory in a function even if you are careful to free it in the calling function? Valgrind didn't detect a memory leak the way I used my function.